### ISSUE TITLE
I can't cache image from gridfs

### ISSUE THREAD

**Issue by duron600** [mdirolf/nginx-gridfs#12]
Hi Mike,
   I want the image from gridfs can be cached in browser, but the image reload every time when I refresh my browser. Could you add any http header like ETag or Last-Modified in response header please? Thanks alot!

**Comment by mdirolf** [comment]
yeah adding an etag might not be a bad idea. i think we should be able to just use the file's md5. could probably also use the _id, since files are read-only, but md5 might be better. anybody have thoughts on this?

**Comment by eltiare** [comment]
Looking at my file documents, it looks like there is an uploadDate property. I'm not sure if my ORM (Mongoid/Carrierwave in Ruby) adds it but if it's universal you can add that as a Last-Modified.  Isn't there some remote chance that different files would get the same md5 hash? I think _id would be better as it is guaranteed to be unique.  Is the md5 hash generated from the contents of the file, from the document itself, or both? If the same file is uploaded twice and the md5 is generated from the file contents but has different document properties that might cause an issue in extreme edge cases (like changing the contentType).

I'm rather partial to the Last-Modified tags, and so if it could be possible if the uploadDate property isn't always available I think it would be a good idea to have an option to specify the field to check for the upload date and then send the proper tags to the client

**Comment by eltiare** [comment]
As a stop gap measure, you can add the expires option to your path. 

<pre>
server {
  listen 80;
  server_name assets.accentuate.me;

  location / {
    expires 1d;
    gridfs db_name field=filename type=string;
  }

}
</pre>

**Comment by duron600** [comment]
I think it will be better if we can configure it, such as using uploadDate for Last-Modified, or using md5 for ETag  :)

**Comment by RFDAJE** [comment]
got any news on this issue, or the mentioned features from @qiumaoyuan had already been implemented in master?

**Comment by LNGi** [comment]
@qiumaoyuan 你找到解决方案方法了吗？我在豆瓣上看到你的帖子了，也找到了这里，囧。。。

**Comment by bladeofsteel** [comment]
Is there any progress in the implementation of this features?

**Comment by LNGi** [comment]
i'm woking on a patch which will make nginx-gridfs outputs an ETag header and a Last-Modified header to the response, i will push this in a few days, i hope you will like it.

**Comment by bladeofsteel** [comment]
Cool news! I wish try this

**Comment by LNGi** [comment]
ok, i added ETag and Last-Modified header suppor for this module, a push request has been make, see https://github.com/mdirolf/nginx-gridfs/pull/36 or https://github.com/liangzhenjing/nginx-gridfs/commit/dc2a95dc3d5e04d2e1815e35c2ef7009e31412bf.

WARN:
i'm not test much, so use it at your own risk, especially when this module seems having memory leak bug. see this issue: https://github.com/mdirolf/nginx-gridfs/issues/34 for more details.
